src/dmd/doc.d

```diff
@@ -117,13 +117,12 @@ extern (C++) class Section
                 "STANDARDS",
                 "THROWS",
                 "VERSION",
-                null
             ];
-            for (size_t i = 0; table[i]; i++)
+            foreach (entry; table)
             {
-                if (icmp(table[i], name, namelen) == 0)
+                if (Port.iequals(entry, name[0 .. namelen]))
                 {
-                    buf.printf("$(DDOC_%s ", table[i]);
+                    buf.printf("$(DDOC_%.*s ", entry.length, entry.ptr);
```

Comment:
When using `.*s`, you must cast the `entry.length` parameter to `int`, because `entry.length` is a `size_t`.
Also, this change to `.*s` doesn't buy anything anyway, since the string is guaranteed to be 0 terminated.

Reply:
I reverted this bit to `.ptr`

Printing is the main issue I encountered when converting pointers to slices, and I think we need a more convenient way to print slices for it to move forward.